src/agent/src/agent.cpp

```diff
@@ -71,14 +65,7 @@ void Agent::Run()
         [this](module_command::CommandEntry& cmd)
         { return DispatchCommand(cmd, m_moduleManager.GetModule(cmd.Module), m_messageQueue); }));
 
-    m_moduleManager.AddModule(Inventory::Instance());
-
-#ifdef ENABLE_LOGCOLLECTOR
-    m_moduleManager.AddModule(Logcollector::Instance());
-#endif
-
-    m_moduleManager.AddModule(m_centralizedConfiguration);
-    m_moduleManager.Setup();
+    m_moduleManager.AddModules();
```

Comment:
I realize now that it would be best to add the modules and enqueue the module manager task before queuing the `CommandsProcessingTask` because the latter needs the former. Picture this, if the `AddModules` and `Start` takes long enough, a command could try to be pushed into an unitialized, or unstarted, module.